src/esp/gfx/replay/Recorder.cpp

```diff
@@ -216,6 +237,33 @@ void Recorder::updateInstanceStates() {
       instanceRecord.recentState = state;
     }
   }
+
+  // Update rig states
+  for (const auto& rigItr : rigNodes_) {
+    int rigId = rigItr.first;
+    for (int boneIdx = 0; boneIdx < rigItr.second.size(); ++boneIdx) {
+      BoneState boneUpdate{};
+      auto* bone = rigItr.second[boneIdx];
+      const auto absTransformMat = bone->absoluteTransformation();
+      auto rotationShear = absTransformMat.rotationShear();
+      // Remove reflection (negative scaling) from the matrix. We assume
+      // constant node scaling for the node's lifetime. It is baked into
+      // instance-creation so it doesn't need to be saved into
+      // RenderAssetInstanceState. See also onCreateRenderAssetInstance.
+      if (rotationShear.determinant() < 0.0f) {
+        rotationShear[0] *= -1.f;
+      }
+
+      Transform absTransform{absTransformMat.translation(),
+                             Magnum::Quaternion::fromMatrix(rotationShear)};
+      boneUpdate.absTransform = absTransform;
+
+      boneUpdate.rigId = rigId;
+      boneUpdate.boneId = boneIdx;
+
+      currKeyframe_.boneUpdates.emplace_back(boneUpdate);
```

Comment:
What is the reason for doing K bone updates instead of a single "rig update" (that updates all K bones)? It looks like your `boneIdx` just goes from 0..numBones, so you could entirely avoid writing those bone indices if you did a single update for all of them, yes?

Reply:
See: https://github.com/facebookresearch/habitat-sim/pull/2120#discussion_r1228735908